server/src/test/scala/org/http4s/server/middleware/DefaultHeadSpec.scala

```diff
@@ -39,14 +44,18 @@ class DefaultHeadSpec extends Http4sSpec {
     }
 
     "allow GET body to clean up on fallthrough" in {
-      var cleanedUp = false
-      val app = DefaultHead(HttpRoutes.of[IO] {
-        case GET -> _ =>
-          val body: EntityBody[IO] = eval_(IO({ cleanedUp = true }))
-          Ok(body)
-      }).orNotFound
-      app(Request[IO](Method.HEAD)).flatMap(_.as[String]).unsafeRunSync()
-      cleanedUp must beTrue
+      for {
+        cleanedUpRef <- Ref[IO].of(false)
+        route = HttpRoutes.of[IO] {
+          case GET -> _ =>
+            val body: EntityBody[IO] = Stream.empty.onFinalizeWeak(cleanedUpRef.set(true))
```

Comment:
We might even be able to test the interruption with `Stream.never.onFinalizeWeak(...)`.